apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/service/appdata/impl/GeneratorServiceImpl.java

```diff
@@ -314,29 +318,32 @@ public void resumeForging() {
     @Override
     public boolean forge(Block lastBlock, int generationLimit, GeneratorMemoryEntity generator)
         throws BlockchainProcessor.BlockNotAcceptedException {
-        long startLog = System.currentTimeMillis();
+        long startLog = timeService.systemTimeMillis();
         int timestamp = generator.getTimestamp(generationLimit);
-        int[] timeoutAndVersion = getBlockTimeoutAndVersion(timestamp, generationLimit, lastBlock);
-        if (timeoutAndVersion == null) {
-            return false;
-        }
-        int timeout = timeoutAndVersion[0];
         if (!verifyHit(generator, lastBlock, timestamp)) {
-            log.debug(this.toString() + " failed to forge at " + (timestamp + timeout) + " height " + lastBlock.getHeight() + " " +
+            log.debug(this.toString() + " failed to forge at " + (timestamp) + " height " + lastBlock.getHeight() + " " +
```

Comment:
Minor issue, let's replace that concatenations to placeholder usage = "{} ... {}... {}..."

Reply:
Fixed